### PR TITLE
fix(rlhf): undo extra GAS divide for DAPO/CISPO/FIPO losses

### LINKED ISSUE #10117: [RLHF] DAPO/CISPO/FIPO 的 loss 在梯度累积下被额外除以 gradient_accumulation_steps
### Checklist / 检查清单
- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述
`loss_type` 为 `dapo`（`cispo`/`fipo` 同样受影响）时，策略 loss 已按整个梯度累积窗口的 completion token 数归一化：

    normalizer = grpo_batch.num_items_in_batch / self.accelerator.num_processes
    loss = (per_token_loss * completion_mask).sum() / normalizer

`num_items_in_batch` 由 `ga_batch_encoded_inputs` 汇总（`swift/rlhf_trainers/grpo_trainer.py:803-809`），覆盖所有 micro-batch × 所有进程，所以单个 micro-batch 只是全局 token 均值的一部分。

但 `transformers.Trainer.training_step` 会在 backward 前再除以 `current_gradient_accumulation_steps`（`Accelerator` 的 `num_steps` 被强制为 1，没有二次缩放），于是 loss 和梯度都被额外缩小 G 倍：期望梯度 g，`gradient_accumulation_steps=4` 时实际只有 g/4。

影响：DAPO 的有效学习率随 `gradient_accumulation_steps` 变化（G 越大更新越弱），日志里的 loss 也被除以 G。

### How to Reproduce / 如何复现
环境：ms-swift main @ 0673cf75、transformers 5.12.1、DeepSpeed ZeRO-3

1. 用 `--loss_type dapo --gradient_accumulation_steps 4` 起 GRPO 训练；
2. 在 `_compute_loss_and_metrics` 打印同一个 micro-batch 的 loss；
3. 参数更新量约为 `gradient_accumulation_steps=1` 时的 1/4。

最小验证：

    loss = torch.tensor(2.0, requires_grad=True)
    (loss / 4).backward()   # 复现 Trainer 的行为：梯度只有目标值的 1/4

### Additional Information / 补充信息
- 相关位置：`swift/rlhf_trainers/grpo_trainer.py:1131-1134`（归一化分支）；transformers `Trainer.training_step`
- 修复 PR：#10116（补偿 `current_gradient_accumulation_steps`，附单测，21 passed）
- `grpo`/`sapo`/`bnpo`/`dr_grpo` 是按 micro-batch 取均值的，不受影响

### PR DESCRIPTION
**Pull request by kabishou11**
# PR type
- [x] Bug Fix
- [ ] New Feature
- [ ] Document Updates
- [ ] More Models or Datasets Support

# PR information

Fixes #10117

## Problem

For `loss_type` in `dapo` / `cispo` / `fipo`, the policy loss is already a slice of the **global token mean** over the whole gradient-accumulation window:

```python
normalizer = grpo_batch.num_items_in_batch / self.accelerator.num_processes
loss = (per_token_loss * completion_mask).sum() / normalizer
```

`num_items_in_batch` is aggregated from `ga_batch_encoded_inputs` (all micro-batches × all processes). A single micro-batch therefore contributes only part of that global mean.

`transformers.Trainer.training_step` still divides the returned loss by `current_gradient_accumulation_steps` before backward (`Accelerator` is created with `num_steps=1`, so the loss is not scaled a second time there). That extra divide is correct for micro-batch-mean losses (`grpo` / `sapo` / `bnpo` / `dr_grpo`), but for these three losses it shrinks both the logged loss and the gradient by GAS.

## Fix

Multiply the window-normalized **policy** loss by GAS so the Trainer divide cancels. Only `cispo` / `dapo` / `fipo`; only when Trainer itself would scale (training, `not model_accepts_loss_kwargs`, `compute_loss_func is None`). Applied before SDAR / CHORD so those auxiliary terms still receive the Trainer `/GAS`. Liger path unchanged.

## Test plan

```text
pytest -q tests/test_align/test_grpo_loss.py
```

CPU only. 34 passed: GAS=1 vs GAS=N gradient match for DAPO/CISPO/FIPO; GRPO/SAPO/BNPO/DR-GRPO not scaled; eval / loss-kwargs / `compute_loss_func` skip compensation.

### CODE REVIEW

**Comment by hjh0119** [comment]
thanks for your contribution, closed in favor of #10116
